k4GaudiPandora/src/DDCaloDigi.cc

```diff
@@ -138,422 +102,382 @@ StatusCode DDCaloDigi::initialize() {
-  cout << "HCAL sc digi:" << endl;
-  _scHcalDigi->printParameters();
-
-  //set up the random engines for ecal and hcal dead cells: (could use a steering parameter though)
-  if (_deadCellEcal_keep) {
-    _randomEngineDeadCellEcal = new CLHEP::MTwistEngine(0, 0);
-  } else {
-    _randomEngineDeadCellEcal = 0;
+    error() << "Could not identify HCAL threshold unit. Please use \"GeV\", \"MIP\" or \"px\"! Aborting." << endmsg;
+    return StatusCode::FAILURE;
   }
 
-  if (_deadCellHcal_keep) {
-    _randomEngineDeadCellHcal = new CLHEP::MTwistEngine(0, 0);
-  } else {
-    _randomEngineDeadCellHcal = 0;
-  }
+  // Set up the scintillator/MPPC digitizer
+  m_scEcalDigi = std::make_unique<DDScintillatorPpdDigi>();
+  m_scEcalDigi->setPEperMIP(m_ecal_PPD_pe_per_mip);
+  m_scEcalDigi->setCalibMIP(m_calibEcalMip);
+  m_scEcalDigi->setNPix(m_ecal_PPD_n_pixels);
+  m_scEcalDigi->setRandomMisCalibNPix(m_ecal_misCalibNpix);
+  m_scEcalDigi->setPixSpread(m_ecal_pixSpread);
+  m_scEcalDigi->setElecNoise(m_ecal_elec_noise);
+  m_scEcalDigi->setElecRange(m_ecalMaxDynMip);
+  debug() << "ECAL sc digi:" << endmsg;
+  m_scEcalDigi->printParameters(debug());
+
+  m_scHcalDigi = std::make_unique<DDScintillatorPpdDigi>();
+  m_scHcalDigi->setPEperMIP(m_hcal_PPD_pe_per_mip);
+  m_scHcalDigi->setCalibMIP(m_calibHcalMip);
+  m_scHcalDigi->setNPix(m_hcal_PPD_n_pixels);
+  m_scHcalDigi->setRandomMisCalibNPix(m_hcal_misCalibNpix);
+  m_scHcalDigi->setPixSpread(m_hcal_pixSpread);
+  m_scHcalDigi->setElecNoise(m_hcal_elec_noise);
+  m_scHcalDigi->setElecRange(m_hcalMaxDynMip);
+  debug() << "HCAL sc digi:" << endmsg;
+  m_scHcalDigi->printParameters(debug());
+
+  return StatusCode::SUCCESS;
 }
 
-retType DDCaloDigi::operator()(
-    const std::map<std::string, const edm4hep::SimCalorimeterHitCollection&>& simCaloHitCollections,
-    const edm4hep::EventHeaderCollection&                                     headers) const {
-  debug() << " process event : " << headers[0].getEventNumber() << " - run  " << headers[0].getRunNumber()
+retType DDCaloDigi::operator()(const edm4hep::SimCalorimeterHitCollection& simCaloHitCollection,
+                               const edm4hep::EventHeaderCollection&       headers) const {
+  debug() << " process event : " << headers[0].getEventNumber() << " - run  " << headers[0].getRunNumber() << endmsg;
 
-          << endmsg;  // headers[0].getRunNumber(),headers[0].getEventNumber()
+  auto col = edm4hep::CalorimeterHitCollection();  // create output CalorimeterHit collection
+  auto Relcol =
+      edm4hep::CaloHitSimCaloHitLinkCollection();  // create relation collection CalorimeterHit-SimCalorimeterHit
 
-  //auto chschcol = edm4hep::CalorimeterHitCollection(); //collection
-  std::map<std::string, edm4hep::CalorimeterHitCollection> outputCollections;
-  auto Relcol = edm4hep::MCRecoCaloAssociationCollection();  // Relation collection CalorimeterHit, SimCalorimeterHit
+  auto uid = m_uidSvc->getUniqueID(headers, name());
```

Comment:
```suggestion
  auto uid = m_uidSvc->getUniqueID(headers.getEventNumber(), headers.getRunNumber(), name());
```

this should make it work with the current release